### ISSUE TITLE
Importing osm failed 

### ISSUE THREAD

**Issue by loveXMTX** [openstreetmap/osm2pgsql#867]
I have installed osm2pgsql on centos6.8,when I used osm2pgsql to import ,the error happened
```
[root@cluster build]# osm2pgsql -d chinaosm -U postgres -P 5432 /state/partition1/pgsql_data/osm_data/china-latest.osm
osm2pgsql version 0.96.0 (64 bit id space)

Using built-in tag processing pipeline
Using projection SRS 3857 (Spherical Mercator)
Osm2pgsql failed due to ERROR: Connection to database failed: invalid connection option "fallback_application_name"
```
I can connect my pgsql server when I don't use osm2pgsql ,why did this error happen?How could I do?

**Comment by pnorman** [comment]
It's odd to get this error, since `fallback_application_name` has been around for a long time. I wonder if we're building the connection string wrong.

To check this, can you set it up to run without `-P`? If you're connecting with psql without specifying it, you shouldn't need it for osm2pgsql.

You also shouldn't let osm2pgsql be the postgres user, but that should be unrelated to the error.

**Comment by lonvia** [comment]
CentOS 6.8 ships with Postgresql 8.4 which does not yet have `fallback_application_name`. So my guess would be that you linked against the system postgres library and not against the one that is shipped with postgresql 10. Make sure to remove all traces of postgresql 8.4 from your system, build osm2pgsql from scratch (including removing the build directory you have completely) and try again.

**Comment by pnorman** [comment]
Yep, if you've compiled against 8.4, that's the problem. 8.4 has been EOL for about 5 years and is not supported by osm2pgsql.

**Comment by loveXMTX** [comment]
> CentOS 6.8 ships with Postgresql 8.4 which does not yet have `fallback_application_name`. So my guess would be that you linked against the system postgres library and not against the one that is shipped with postgresql 10. Make sure to remove all traces of postgresql 8.4 from your system, build osm2pgsql from scratch (including removing the build directory you have completely) and try again.

It seems to be the reason,I did not remove pgsql 8.4 when I compiled osm2pgsql .But after I got this error ,I removed the old version 8.4,and then used osm2pgsql to try again,the error still existed,so I installed osm2pgsql on windows to import osm data,that was successful.Is it because I hadn't complied osm2pgsql again? Since I update my system to Centos7 ,I cannot try.
